### PR TITLE
Added --stepsize parameter

### PR DESCRIPTION
**Pull request by EklavyaFCB**
Addressing #88 

Updated `ultraLatency.py` file to enable the parameter --stepSize. 

This is was a straightforward implementation and required only:
- Adding another `parser.add_option()` with the necessarily information
- Adding the `options.stepSize` value in the configureScanModule function. The original function in cmsgemos/gempython/tools already had the required stepsize variable input.

Tested the updated script and can confirm with the following plots:
http://cmsonline.cern.ch/cms-elog/997121

Could/should add an if-statement in case the user inputs a stepsize larger than the difference of latencies (scanmax-scanmin).

Usage example (should add more parameters for a real scan):

`ultraLatency.py --scanmin=0 --scanmax=100 --stepSize=10`

Creates the `LatencyData_Trimmed.root` as normal.

### CODE REVIEW

**Comment by EklavyaFCB** [comment]
@bdorney Here is the Pull Request of the stepsize feature implementation. Hope all is good.

**Comment by bdorney** [comment]
Please add a check on stepSize:

> Could/should add an if-statement in case the user inputs a stepsize larger than the difference of latencies (scanmax-scanmin).

If:

`scanmin+stepSize > scanmax`

Then have the scan only execute the scan for scanmin and scanmax.  Explicitly change stepsize to be the difference between the two:

`stepSize = scanmax - scanmin`

**Comment by EklavyaFCB** [comment]
Added

File: ultraLatency.py
Comment (bdorney) [inline ultraLatency.py]: You should not try to change options.stepSize as this is cmd line entry from the user and may be required elsewhere.

Instead you should store options.stepSize as a local variable and then modify this variable accordingly

**Comment by EklavyaFCB** [comment]
Modified.

Also added an else-statement in case the user doesn’t input a --stepwise parameter to not get an error in the configureScanModule function.

File: ultraLatency.py
Comment (bdorney) [inline ultraLatency.py]: Please make stepSize have a default value, set equal to 1.

File: ultraLatency.py
Comment (bdorney) [inline ultraLatency.py]: This case structure at line 57 is not needed.  Remove, only the case structure at line 59-60 should be used.   

Additionally you need to assign the value of step before the case structure, otherwise you will have a scope issue, e.g. at line 193 `step=0` because of this scope issue.

File: ultraLatency.py
Comment (cbravo135) [inline ultraLatency.py]: There is no reason for this check if you have a default value. What is it you are trying to check here anyway?

Comment (cbravo135) [reply ultraLatency.py]: I would check step > 0 and that seems like the only thing you need. We also should propogate this change into run_scans.py before we can merge this change.

**Review by cbravo135 — CHANGES_REQUESTED** [review]
Please propagate this change to include this option in run_scans.py

File: ultraLatency.py
Comment (cbravo135) [inline ultraLatency.py:62]: why have this line?

File: ultraLatency.py
Comment (cbravo135) [inline ultraLatency.py]: Just use options.stepSize directly here, you only use it once...

**Review by cbravo135 — DISMISSED** [review]
Please propagate the changes to run_scans.

File: ultraLatency.py
Comment (EklavyaFCB) [reply ultraLatency.py]: Yes, but we want the value of stepsize to be at max equal to scanmax-scanmin. So we assign to a variable the value of options.stepsize, rather than modify options.stepsize directly as the latter is a command line entry and could be required elsewhere as Brian pointed out. 

Therefore in the function, the variable with the updated value should be used, not whats input in the command line.

Comment (cbravo135) [reply ultraLatency.py]: Sure, this is not nearly as important as this change being propagated to run_scans. We can not approve this PR until this happens.

Comment (EklavyaFCB) [reply ultraLatency.py]: Yeah, sure, I didn't mean for all the commits to send PR notifications every time. Just needed to update my repo to download it in a different location. 

I implemented some changes run_scans.py but couldn't test them yet (my $DATA_PATH didn't have permissions yet). Soon ...

Comment (bdorney) [reply ultraLatency.py]: `$DATA_PATH` is a `$SHELL` variable that you set.  You can change the directory it is set too, and based on my earlier comments you should.

**Review by EklavyaFCB — COMMENTED** [review]
There were some merge conflicts with my version of the files vs the ones Brian updated in the last few days. Have solved the conflicts so the merge shouldn't be an issue.

File: ultraLatency.py
Comment (EklavyaFCB) [reply ultraLatency.py:62]: For lines 56-58: in case the user inputs a negative stepSize, we should print an error line, like with scanmin/max, MSPL and vt2 above.

For line 60: as mentioned earlier, to use a variable of hold the value of stepSize rather than using the direct CLI value, in order to calculate the difference between the scanmax and min if necessary.

File: run_scans.py
Comment (bdorney) [inline run_scans.py]: These lines in run_scans.py are unnecessary. You already coded this check in ultraLatency.py.

You should only be passing the parameter here

File: run_scans.py
Comment (bdorney) [inline run_scans.py]: What is the default value that stepSize will take?

Ensure that it is 1.

**Review by bdorney — CHANGES_REQUESTED** [review]
Changes requested. Cross check what default value is being passed to launchArgs for stepSize.

Additionally launchArgs should only pass parameters to the tool of interest; so please revise lines 110-115 of run_scans.py

File: run_scans.py
Comment (bdorney) [inline run_scans.py]: `stepSize` is not boolean but integer, example, how is `stepSize = 20` treated?  Please change to something like `stepSize > 0`

Also `step` looks like it has the value of 0 here, why use another variable at all? You have `stepSize`, use that.

File: qcoptions.py
Comment (bdorney) [inline qcoptions.py]: Right now are run_scans.py and ultraLatency.py the only ones that use the `stepSize` option?  I think yes, please keep `stepSize` option hard coded in only these two scripts to avoid confusion.  For example if you call:

`ultraThreshold.py -h`

You'll see the `stepSize` option right now but it doesn't do anything in `ultraThreshold.py.`

Please update.

Also, keep the options menu's alphabetized for readability.

**Review by bdorney — CHANGES_REQUESTED** [review]

File: run_scans.py
Comment (EklavyaFCB) [reply run_scans.py]: Ack, I did have options.stepSize there, but b/c of the git push CLI issues, I updated to code on the web GUI, and missed that one.

**Comment by bdorney** [comment]
Please organize time for testing on the cosmic stand then upload the elog short link of the results here

**Comment by EklavyaFCB** [comment]
Tests for run_scans.py and ultraLatency.py can be seen here:

http://cmsonline.cern.ch/cms-elog/1000339

The only small remaining issue is the `[None]` that gets printed before the `Normal termination`. It creates the .root files in `GE11-VI-L-CERN-0001/latency/trk/current` fine though, then you just have to run the analysis on them, using anaUltraLatency.py -i in this case with the correct path of the .root file.

**Comment by EklavyaFCB** [comment]
The last commit was for a different pull request (#86) that I'd prepared. Just saw that you can't make another PR until a previous one is fully approved and merged. In which case I can remove the changes in the latest commit if required. Please let me know.

**Comment by jsturdy** [comment]
See [this](http://nvie.com/posts/a-successful-git-branching-model/) for a description of the branching model that we try to use (though not as rigorously in this repo yet... (not to name names)
Basic outline:
* Start from your branch which is up-to-date with the upstream branch you're developing a feature for
* Create a new branch `my-great-new-feature`
* Add commits to *this* feature branch related to the feature you are developing
* Push this branch to your remote
* Create a PR from `my-great-new-feature` to the upstream branch that your feature is intended for

If you happen to be developing two independent features, you do the exact same thing for the second feature.  In this way, you won't have the issue that you're encountering now, and all PRs will be more cleanly managed.
There may be problems if your two features depend on one another, but this can easily be accounted for by making your dependent feature a new branch off of your feature development branch

**Comment by EklavyaFCB** [comment]
Thanks Jared, appreciate the advice. Have removed the updates in the latest commit, so this PR should be good to go based on the e-log I posted earlier.

For #86 I'll create a new branch and add the commits to that one, then PR to the upstream branch from there.

File: qcoptions.py
Comment (bdorney) [inline qcoptions.py]: are these tab characters (e.g. \t) here?  If so please change your editor such that when pressing "tab" a fixed number of spaces are inserted.  If you use vim this can be accomplished by editing:

`~/.vimrc`

And placing the following lines:

`
" show existing tab with 4 spaces width
set tabstop=4
" " when indenting with '>', use 4 spaces width
set shiftwidth=4
" " On pressing tab, insert 4 spaces
set expandtab
`

Into the file.

Otherwise this is fine for me.

File: qcoptions.py
Comment (bdorney) [inline qcoptions.py]: There is still a tab character here, e.g.  "\t"

**Review by bdorney — CHANGES_REQUESTED** [review]

**Comment by EklavyaFCB** [comment]
Had to remove an extra space elsewhere (in MSPL) for git to detect the tab removal for --vfatmask.

`cat -T qcoptions.py` showed no tab character.

**Review by bdorney — COMMENTED** [review]
Can you confirm that this PR is ahead of the master branch? e.g. you do not have unmerged changes between your branch and the central remote's master branch?

**Comment by EklavyaFCB** [comment]
My branch is up to date with the central repo:

`From github.com:cms-gem-daq-project/vfatqc-python-scripts`
`*branch            master     -> FETCH_HEAD`
`Already up-to-date.`

**Review by bdorney — APPROVED** [review]

**Pull request merged**
